gluoncv/model_zoo/icnet.py

```diff
@@ -0,0 +1,381 @@
+
+        # sub 4
+        x_sub4 = F.contrib.BilinearResize2D(x_sub2_out,
+                                            height=x_sub2_out.shape[2] // 2,
+                                            width=x_sub2_out.shape[3] // 2)
+
+        x = self.layer3(x_sub4)
+        x = self.layer4(x)
+        x_sub4_out = self.psp_head(x)
+
+        x_sub4_out = self.conv_sub4(x_sub4_out)
+        x_sub2_out = self.conv_sub2(x_sub2_out)
+
+        res = self.head(x_sub1_out, x_sub2_out, x_sub4_out)
+
+        return res[0]
+
+
+class ConvBnRelu(HybridBlock):
+    def __init__(self, in_planes, out_planes, ksize, stride=1, pad=0, dilation=1,
+                 groups=1, has_bn=True, norm_layer=nn.BatchNorm, bn_eps=1e-5,
+                 has_relu=True, has_bias=False):
+        super(ConvBnRelu, self).__init__()
+        with self.name_scope():
+            self.conv = nn.Conv2D(in_channels=in_planes, channels=out_planes, kernel_size=ksize,
+                                  padding=pad, strides=stride,
+                                  dilation=dilation, groups=groups, use_bias=has_bias)
+            self.has_bn = has_bn
+            self.has_relu = has_relu
+
+            if self.has_bn:
+                self.bn = norm_layer(in_channels=out_planes, epsilon=bn_eps)
+            if self.has_relu:
+                self.relu = nn.Activation('relu')
+
+    def hybrid_forward(self, F, x):
+        x = self.conv(x)
+        if self.has_bn:
+            x = self.bn(x)
+        if self.has_relu:
+            x = self.relu(x)
+
+        return x
+
+    def demo(self, x):
+        x = self.conv(x)
+        if self.has_bn:
+            x = self.bn(x)
+        if self.has_relu:
+            x = self.relu(x)
+
+        return x
+
+
+class CascadeFeatureFusion(HybridBlock):
+    def __init__(self, low_channels, high_channels, out_channels, nclass, norm_layer=nn.BatchNorm):
+        super(CascadeFeatureFusion, self).__init__()
+
+        with self.name_scope():
+            self.conv_low = nn.HybridSequential()
+            with self.conv_low.name_scope():
+                self.conv_low.add(nn.Conv2D(in_channels=low_channels, channels=out_channels,
+                                            kernel_size=3, padding=2, dilation=2, use_bias=False))
+                self.conv_low.add(norm_layer(in_channels=out_channels))
+
+            self.conv_hign = nn.HybridSequential()
+            with self.conv_hign.name_scope():
+                self.conv_hign.add(nn.Conv2D(in_channels=high_channels, channels=out_channels,
+                                             kernel_size=1, use_bias=False))
+                self.conv_hign.add(norm_layer(in_channels=out_channels))
+
+            self.conv_low_cls = nn.Conv2D(in_channels=out_channels, channels=nclass,
+                                          kernel_size=1, use_bias=False)
+
+    def hybrid_forward(self, F, x_low, x_high):
+        x_low = F.contrib.BilinearResize2D(x_low, height=x_high.shape[2], width=x_high.shape[3])
```

Comment:
same thing here, pass in the shape instead of getting a shape inside hybrid_forward.